src/Microsoft.AspNetCore.Blazor/Json/SimpleJson/SimpleJson.cs

```diff
@@ -1430,7 +1430,13 @@ public virtual object DeserializeObject(object value, Type type)
                             obj = value;
                         else
                         {
-                            obj = ConstructorCache[type]();
+                            try{
+                                obj = ConstructorCache[type](); 
+                            }
+                            catch(NullReferenceException)
+                            {
+                                throw new InvalidOperationException($"Objects must have a constructorless parameter to deserialize. Offending type '{type}'");
```

Comment:
Could we change the message to the following format?

    Cannot deserialize JSON into type '{type.FullName}' because it does not have a public parameterless constructor.